components/raftstore/src/store/peer.rs

```diff
@@ -2550,7 +2567,40 @@ where
                 // Update apply index to `last_applying_idx`
                 self.read_progress
                     .update_applied(self.last_applying_idx, &ctx.coprocessor_host);
-                self.notify_leader_the_peer_is_available(ctx);
+                if self.wait_data {
```

Comment:
Question: Why do we need to call apply_gap after the snapshot is applied for wait_data == true? I mean why not catch up the gaps just like normal cases (apply snapshot and then apply new logs since the last applied index)

Reply:
For normal cases, `ready.committed_messages` have all been processed before `apply snapshot`. But for this case, in order to avoid the side effects of continuing to apply other cmds after applying the `non-witness switch`, the logs behind `non-witness switch` will be skipped directly. So we need to make sure the skipped logs will be all applied.